### ISSUE TITLE
urgent: how to evaluate during training 

### ISSUE THREAD

**Issue by juliahane** [allenai/allennlp#3254]
I was wondering if you could tell me how I can do the evaluation during the training: so instead of doing allennlp train then evaluate, do the evaluation and train with one command.
thanks

**Comment by joelgrus** [comment]
you need to do two things:

1. add a `test_data_path` path to your config file (similar totrain_data_path and validation_data_path)
2. add `evaluate_on_test: true` to your config file (again, at the top level)

note that we typically discourage this workflow; from a "good science" perspective you want to evaluate on the test set as little as possible (and in particular not programmatically), so that you don't end up overfitting to the test set. 

(this is why it requires an explicit `evaluate_on_test` key)
